autoload/go/lint.vim

```diff
@@ -18,17 +14,23 @@ if !exists("g:go_errcheck_bin")
     let g:go_errcheck_bin = "errcheck"
 endif
 
-function! go#lint#Gometa(path_to_lint) abort
+function! go#lint#Gometa(...) abort
+    if a:0 == 0
+        let goargs = expand('%:p:h')
```

Comment:
None of the other linters do recursive linting by default, why is GoMetaLinter a special case?  There's no option to tell gometalinter to output full paths, so that's not a possible solution.

Passing a single file to gometalinter is not valid input, you must pass it a path (`./...` is a path).  You should still be able to call `GoMetaLinter ./...` to get that behaviour, like some of the other linters.

I don't like the idea of forcing people to add stuff to their vimrc to get default behaviour working, and changing the working directory by default will produce unintended behaviour for people.